djangocms_admin_style/static/js/drag-touch-support.js

```diff
@@ -1,33 +1,35 @@
+if (window.jQuery) {
+    (function ($) {
+        $.fn.touchSupport = function () {
+            function touchHandler(event) {
+                var touch = event.changedTouches[0];
+                var simulatedEvent = document.createEvent('MouseEvent');
 
-(function ($) {
-    $.fn.touchSupport = function(){
-        function touchHandler(event) {
-            var touch = event.changedTouches[0];
-            var simulatedEvent = document.createEvent('MouseEvent');
+                simulatedEvent.initMouseEvent({
+                        touchstart: 'mousedown',
+                        touchmove: 'mousemove',
+                        touchend: 'mouseup'
+                    }[event.type], true, true, window, 1,
+                    touch.screenX, touch.screenY,
+                    touch.clientX, touch.clientY, false,
+                    false, false, false, 0, null);
 
-            simulatedEvent.initMouseEvent( {
-                touchstart: 'mousedown',
-                touchmove: 'mousemove',
-                touchend: 'mouseup'
-            }[event.type], true, true, window, 1,
-                touch.screenX, touch.screenY,
-                touch.clientX, touch.clientY, false,
-                false, false, false, 0, null);
-
-            touch.target.dispatchEvent(simulatedEvent);
-            event.stopPropagation();
-        }
+                touch.target.dispatchEvent(simulatedEvent);
+                event.stopPropagation();
+            }
 
-        function init() {
-            var that = $(this);
-            for (var i = 0; i < $(this).length; i++) {
-                that[i].addEventListener('touchstart', touchHandler, true);
-                that[i].addEventListener('touchmove', touchHandler, true);
-                that[i].addEventListener('touchend', touchHandler, true);
-                that[i].addEventListener('touchcancel', touchHandler, true);
+            function init() {
+                var that = $(this);
+                for (var i = 0; i < $(this).length; i++) {
```

Comment:
Better to save length outside, it will be much more faster.
https://stackoverflow.com/questions/17989270/for-loop-performance-storing-array-length-in-a-variable